### ISSUE TITLE
topology manager presubmit jobs seems to be skipping all the tests

### ISSUE THREAD

**Issue by ffromani** [kubernetes/kubernetes#125950]
The topology manager presubmit jobs seems to be skipping, not running, all the actual tests. [Reproducer PR](https://github.com/kubernetes/kubernetes/pull/125919/commits/2a76171f6dfaa2b9d18b9f20938037c149634089), on which I add a `ginkgo.Fail` in these tests and trigger them with:
```
/test pull-kubernetes-node-kubelet-serial-topology-manager
/test pull-kubernetes-node-kubelet-serial-topology-manager-kubetest2
```
yet the PR is passing the above lanes:
![image](https://github.com/kubernetes/kubernetes/assets/25349329/93d0ef0e-e2fb-4563-ac0e-300bf3806bab)

for /test pull-kubernetes-node-kubelet-serial-topology-manager"
![image](https://github.com/kubernetes/kubernetes/assets/25349329/8fdb0465-5977-42d0-b92b-490acdfb1b18)

for /test pull-kubernetes-node-kubelet-serial-topology-manager-kubetest2
![image](https://github.com/kubernetes/kubernetes/assets/25349329/b5f7afd2-a665-4760-8180-7720212d9c41)

local run on which test do fail as expected:
```
make test-e2e-node FOCUS='Feature:TopologyManager' SKIP='' VERBOSE=3 GINKGO_ARGS='-gingko.v -test.v'
```
![image](https://github.com/kubernetes/kubernetes/assets/25349329/57db0f92-3ad2-437c-b342-d84929b07c19)

**Comment by ffromani** [comment]
using `-label-filter=TopologyManager` doesn't seem to help

**Comment by pohly** [comment]
I probably have  fix for that in https://github.com/kubernetes/kubernetes/pull/125163 where I noticed the same for the DRA presubmit. As it affects other jobs, let's merge that fix separately: https://github.com/kubernetes/kubernetes/pull/125966/files

**Comment by ffromani** [comment]
Thanks @pohly for the quick fix. Let me verify ASAP.

**Comment by ffromani** [comment]
/reopen

the `-kubetest2` lane seems to be skipping, but #125966 helps a lot, we have a functional job again

**Comment by SergeyKanzhelev** [comment]
@ffromani is this one still needs to be open?

**Comment by ffromani** [comment]
> @ffromani is this one still needs to be open?

yes, I'll try to wrap up after the enhancement freeze

**Comment by SergeyKanzhelev** [comment]
/remove-lifecycle rotten

@ffromani is it still an issue?

**Comment by ffromani** [comment]
/remove-priority important-longterm
/priority important-soon

I was wrong, still an issue even though "only" the `-kubetest2` lane is broken
```
[ReportAfterSuite] Kubernetes e2e JUnit report
k8s.io/kubernetes/test/e2e/framework/test_context.go:615
[ReportAfterSuite] PASSED [0.004 seconds]
------------------------------

Ran 0 of 740 Specs in 0.012 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 740 Skipped
```

I think this is my next top priority

**Comment by ffromani** [comment]
turns out the `kubetest2` `node` tester, which we need to use to run the `e2e_node` suite, includes the `Serial` qualifier among the default skips if nothing is given explicitely. Which was exactly the case for the affected lane. See: https://github.com/kubernetes-sigs/kubetest2/blob/master/pkg/testers/node/node.go#L92

https://github.com/kubernetes/test-infra/pull/34563 should fix this.

**Comment by pohly** [comment]
This default kubetest2 skip has bitten various people over time, including @bart0sh and on another occasion myself (when converting to --label-filter). I don't think it should be opinionated like that, it's totally non-obvious when looking at a job definition.

@BenTheElder, @aojea: what do you think about removing that default? It is a breaking change, though.

**Comment by BenTheElder** [comment]
> turns out the kubetest2 node tester, which we need to use to run the e2e_node suite, includes the Serial qualifier among the default skips if nothing is given explicitely. Which was exactly the case for the affected lane

Yes, because by default we run test in parallel? `[Serial]` is supposed to denote a test that cannot be run safely in parallel with other tests.

These tests get run in some periodic CI, but running every test serially is too slow for presubmit.

> @BenTheElder, @aojea: what do you think about removing that default? It is a breaking change, though.

Not skipping serial by default seems questionable, however if we're only talking about node tester, I have less strongly held opinions :-)

**Comment by pohly** [comment]
And I can imagine that folks who specifically set up a job with the intend to run serial, slow and disruptive tests will be left scratching their heads why those tests don't run.

**Comment by BenTheElder** [comment]
> And I can imagine that folks who specifically set up a job with the intend to run serial, slow and disruptive tests will be left scratching their heads why those tests don't run.

Yeah that's fair, but I'm concerned about breaking ~everyone else, since this behavior is ... pretty old.

**Comment by pohly** [comment]
We want people to migrate to kubetest2 and --label-filter also for e2e jobs. We probably need to warn about this gotcha as part of the communication. I think I had it in my email draft.

**Comment by pohly** [comment]
One option would be to disable the skip default when --label-filter is used. --label-filter support in kubetest2 is new, so we wouldn't break existing jobs.

**Comment by ffromani** [comment]
This was already covered by previous comments, so fully acknowledging the risk of being obvious, but still, mostly for historical record:

Q: Why we use `Serial` quite a lot in the `e2e_node` suite, so this bite us?
A: there's a bunch of tests in the `e2e_node` suite which are `Serial` because they need to mutate and check the precise state of the node, which is (the node) a big blob of implicit shared state. At minimum we need to precisely control how many pods are running in order to make assertions (ginkgo expectations). Then there is the resource (CPU, device, memory) allocation, and the list goes on.

Not sure there's a better/simpler way which enables parallel testing, but has not emerged yet.

**Comment by pohly** [comment]
> Not sure there's a better/simpler way which enables parallel testing, but has not emerged yet.

You can simply label your serial test with `WithSerial` and then run the suite with `ginkgo -p`. Ginkgo v2 then automatically runs parallel tests in parallel and the serial tests sequentially:

https://pkg.go.dev/k8s.io/kubernetes/test/e2e/framework#WithSerial

**Comment by ffromani** [comment]
/close

remaining bits fixed by
https://github.com/kubernetes/test-infra/pull/34590
https://github.com/kubernetes/test-infra/pull/34563
